File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        L={}
        for i in range(len(nums)):
            X=target-nums[i]
            if X in L:
                return [L[X],i]
            L[nums[i]]=i
        return False

File: Data Structures & Algorithms/rotate-array/submission-1.java
class Solution {
    public void rotate(int[] nums, int k) {
        k = k % nums.length;
        rotate(nums, 0, nums.length-1);
        rotate(nums, 0, k-1);
        rotate(nums, k, nums.length-1);
    }
    private void rotate(int[] nums, int start, int last){
        while(start<last){
            int temp = nums[last];
            nums[last] = nums[start];
            nums[start] = temp;
            start++;
            last--;
        }
    }
}